semantics/semBakeryScript.sml

```diff
@@ -0,0 +1,67 @@
+open preamble basis
+
+open astBakeryTheory
+
+val _ = new_theory "semBakery";
+
+val freeprocs_def = Define`
+  freeprocs (LTau p)           = {p}
+∧ freeprocs (LCom p1 v1 p2 v2) = {p1;p2}
+∧ freeprocs (LSel p1 b p2)     = {p1;p2}
+`;
+
+val sender_def = Define`
+  sender (LTau p)           = NONE
+∧ sender (LCom p1 v1 p2 v2) = SOME p1
+∧ sender (LSel p1 b p2)     = SOME p1
+`;
+
+val receiver_def = Define`
+  receiver (LTau p)           = NONE
+∧ receiver (LCom p1 v1 p2 v2) = SOME p2
+∧ receiver (LSel p1 b p2)     = SOME p2
+`;
+
+val (trans_rules,trans_ind,trans_cases) = Hol_reln `
+(∀s v1 p1 v2 p2 d c.
+    FLOOKUP s (v1,p1) = SOME d
+    ⇒ trans (s,Com p1 v1 p2 v2 c) (LCom p1 v2 p2 v2) (s |+ ((v2,p2),d),c))
+∧ (∀s p1 b p2 c.
+    trans (s,Sel p1 b p2 c) (LSel p1 v p2) (s,c))
+∧ (∀s v p f vl c.
+    EVERY IS_SOME (MAP (FLOOKUP s) (MAP (λv. (v,p)) vl))
+    ⇒ trans (s,Let v p f vl c) (LTau p) (s |+ ((v,p),f(MAP (THE o FLOOKUP s) (MAP (λv. (v,p)) vl))),c))
+∧ (∀s v p c1 c2.
+    FLOOKUP s (v,p) = SOME [1w]
+    ⇒ trans (s,IfThen v p c1 c2) (LTau p) (s,c1))
+(* Swapping transitions *)
+∧ (∀s v p c1 c2 s' c1' c2'.
+    trans (s,c1) alpha (s',c1')
+
+    ∧ trans (s,c2) alpha (s',c2')
+    ∧ p ∉ freeprocs alpha
+    ⇒ trans (s,IfThen v p c1 c2) alpha (s',IfThen v p c1' c2'))
+∧ (∀s c s' c' p1 v1 p2 v2.
+    trans (s,c) alpha (s',c')
+    ∧ p1 ∉ freeprocs alpha
```

Comment:
@IlmariReissumies Do we really need `p1` to be in `freeprocs alpha` ? In Figure 2.3 they only check for `p2`

Reply:
The rule you are looking at is meant to mimic the swapping relation of Figure 2.2, where they do indeed check for disjointness of the processes. The asynch rule in Figure 2.3 corresponds to the bottommost rule in my definition.